dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1543,10 +1549,13 @@ public String getTableToolContent(@ApiParam(hidden = true) @Parameter(hidden = t
         if (mainDescriptor != null) {
             Set<SourceFile> secondaryDescContent = extractDescriptorAndSecondaryFiles(workflowVersion);
             LanguageHandlerInterface lInterface = LanguageHandlerFactory.getInterface(workflow.getFileType());
-            final String toolTableJson = lInterface.getContent(workflowVersion.getWorkflowPath(), mainDescriptor.getContent(), secondaryDescContent,
+            final Optional<String> toolTableJson = lInterface.getContent(workflowVersion.getWorkflowPath(), mainDescriptor.getContent(), secondaryDescContent,
                 LanguageHandlerInterface.Type.TOOLS, toolDAO);
-            workflowVersion.setToolTableJson(toolTableJson);
```

Comment:
Before, if there was an error, it would `setToolTableJson(null)`. Now you don't set it at all. Do you want that?

I don't know the answer, just raising the question. In theory, without understanding the whole flow, it seems like if there were already a value in that fiel, it would get cleared before your change, and now it doesn't.

This happens in at least 2 other places (ran across this one first). Checking that this is as intended everywhere. Could very well be the bug fix :) But I'm still concerned about old values not being cleared.

Reply:
Oh good point, thinking about it more, it does make sense to clear it in this instance and here https://github.com/dockstore/dockstore/pull/3794/files/9f19334385fe1fd251b7189ca3938b378cb411fd#diff-f9e0ac900016e3693ac10f2a3a7b8939L1505 too.
Unsure of the other location you were referring to.